pkg/cloud/metadata.go

```diff
@@ -69,13 +75,69 @@ func (m *Metadata) GetOutpostArn() arn.ARN {
 func NewMetadata() (MetadataService, error) {
 	sess := session.Must(session.NewSession(&aws.Config{}))
 	svc := ec2metadata.New(sess)
-	return NewMetadataService(svc)
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
```

Comment:
I think we shouldonly do this if metadata is unavailable.

Reply:
you're right. i'll change lines 80 and 85 to
`if err != nil && !svc.Available() {`